app/src/main/java/com/bignerdranch/android/criminalintent/CrimeDetailViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.bignerdranch.android.criminalintent
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import java.util.UUID
+
+class CrimeDetailViewModel(): ViewModel() {
+
+    private val crimeRepository = CrimeRepository.get()
+    private val crimeIdLiveData = MutableLiveData<UUID>()
+
+    var crimeLiveData: LiveData<Crime?> =
+        Transformations.switchMap(crimeIdLiveData) { crimeId ->
+            crimeRepository.getCrime(crimeId)
+        }
```

Comment:
It is a good practice in OOP languages ([encapsulation](https://www.sumologic.com/glossary/encapsulation/)) to restrict access to mutable variables in order to improve state management, however public variables should be immutable. So, in the example above, the `crimeLiveData` should be a `val` instead of `var`. That way, you can never change it's value, _unless_ you change the value of `crimeIdLiveData` which is a `MutableLiveData` type, because the result of `crimeLiveData` is directly affected by the value of `crimeIdLiveData`.

Reply:
Good point! However according to the book 'crimeLiveData' should be 'var' so I'm sticking to the book.